server/auth/sso/sso.go

```diff
@@ -163,14 +192,49 @@ func newSso(
 		}
 	}
 
-	clientID := clientIDObj.Data[c.ClientID.Key]
-	if clientID == nil {
-		return nil, fmt.Errorf("key %s missing in secret %s", c.ClientID.Key, c.ClientID.Name)
+	return privateKey, nil
+}
+
+func newSso(
+	factory providerFactory,
+	c Config,
```

Comment:
we should probably keep these references consistently named as `c` or `config`, not both